assets/js/components/surveys/SurveyViewTrigger.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { useMount } from 'react-use';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+
+const { useSelect, useDispatch } = Data;
+
+const SurveyViewTrigger = ( { triggerID, ttl } ) => {
+	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
+
+	const { triggerSurvey } = useDispatch( CORE_USER );
+
+	useMount( () => {
+		if ( usingProxy ) {
+			triggerSurvey( { triggerID, ttl } );
+		}
+	} );
+
+	return null;
+};
+
+SurveyViewTrigger.propTypes = {
+	triggerID: PropTypes.string.isRequired,
+	ttl: PropTypes.number,
+};
+
+SurveyViewTrigger.defaultProps = {
+	ttl: null,
```

Comment:
The default value for the `ttl` should be `0`.

Reply:
So I don't read the AC or IB like that @eugene-manuilov  *BUT* I can totally see that from following other PRs

You may have an extra comment on how I do default values and destructuring though so I'm expecting this to be sent back again :smile: